src/main/java/io/openliberty/tools/intellij/lsp4jakarta/lsp4ij/codeAction/JakartaCodeActionHandler.java

```diff
@@ -80,8 +82,7 @@ public List<CodeAction> codeAction(JakartaJavaCodeActionParams params, IPsiUtils
             var mpParams = new MicroProfileJavaCodeActionParams(params.getTextDocument(), params.getRange(), params.getContext());
             // We need to clone the contents of the editor and rebuild the PSI so that we can modify it in our quick fixes.
             // If we do not then we receive a write access exception because we are in a runReadAction() context.
-            JavaCodeActionContext context = new JavaCodeActionContext(unit.getViewProvider().clone().getPsi(unit.getLanguage()),
-                    start, end - start, utils, mpParams);
+            JavaCodeActionContext context = new JavaCodeActionContext(unit, start, end - start, utils, mpParams).copy();
```

Comment:
This change is to ensure that the source JavaCodeActionContext (returned by context.getSource()) always contains an unmodified PsiFile.

Reply:
👍